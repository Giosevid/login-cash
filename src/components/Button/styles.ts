import {StyleSheet} from 'react-native';
import colors from '../../assets/colors';

const styles = StyleSheet.create({
  button: {
    alignItems: 'flex-end',
    marginTop: 30,
  },
  label: {
    color: colors.secondary,
    fontWeight: 'bold',
  },
  wrapper: {
    width: 150,
    height: 40,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 50,
    flexDirection: 'row',
  },
});

export default styles;
