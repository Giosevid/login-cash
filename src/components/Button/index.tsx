import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';
import {
  NavigationDrawerProps,
  RootStackParamList,
} from '../../navigation/interfaces/navigation';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import {useNavigation} from '@react-navigation/native';
import styles from './styles';
import colors from '../../assets/colors';

type Props = {
  label: string;
  route: keyof RootStackParamList;
};

const Button = ({label, route}: Props) => {
  const navigation = useNavigation<NavigationDrawerProps>();
  return (
    <View style={styles.button}>
      <TouchableOpacity onPress={() => navigation.navigate(route)}>
        <LinearGradient colors={['#031f34', '#123c38']} style={styles.wrapper}>
          <Text style={styles.label}>{label}</Text>
          <MaterialIcons
            name="navigate-next"
            color={colors.secondary}
            size={20}
          />
        </LinearGradient>
      </TouchableOpacity>
    </View>
  );
};

export default Button;
