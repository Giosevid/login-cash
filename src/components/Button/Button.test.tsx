import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Button from './index';

jest.mock('react-native-linear-gradient', () => 'LinearGradient');
jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon');

describe('Test Button', () => {
  const handleClick = jest.fn();
  test('Snapshot Button', () => {
    const {toJSON} = render(
      <Button onPress={handleClick} label="Test Button" />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('Expect call handleClick', () => {
    const {getByText} = render(
      <Button onPress={handleClick} label="Test Button" />,
    );
    const element = getByText('Test Button');
    fireEvent.press(element);
    expect(handleClick).toBeCalledTimes(1);
  });
});
