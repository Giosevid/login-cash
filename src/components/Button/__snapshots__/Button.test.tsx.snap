// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Button Snapshot Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 16,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <LinearGradient
    colors={
      Array [
        "#7474BF",
        "#348AC7",
        "#7474BF",
      ]
    }
    end={
      Object {
        "x": 1,
        "y": 0,
      }
    }
    start={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#87ff2f",
          "fontWeight": "bold",
        }
      }
    >
      Test Button
    </Text>
    <Icon
      color="#87ff2f"
      name="navigate-next"
      size={20}
    />
  </LinearGradient>
</View>
`;
